reactor-core/src/test/java/reactor/core/publisher/FluxFirstWithValueTest.java

```diff
@@ -249,6 +249,12 @@ void combineMoreThanOneAdditionalSource() {
 						.doesNotContainNull());
 	}
 
+	// See https://github.com/reactor/reactor-core/issues/2557
+	@Test
+	void protectAgainstSparseArray() {
+		Flux.firstWithValue(Arrays.asList(Mono.just(1), Mono.empty())).blockFirst();
```

Comment:
maybe be more explicit, like using either a `StepVerifier` or an AssertJ `assertThatCode.doesNotThrowAnyException(() -> {...})` ?